lib/wallet/src/Cardano/Wallet/Write/Tx/Balance/CoinSelection.hs

```diff
@@ -0,0 +1,693 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE RoleAnnotations #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{- HLINT ignore "Use <$>" -}
+{- HLINT ignore "Use camelCase" -}
+
+-- |
+-- Copyright: © 2023 IOHK
+-- License: Apache-2.0
+--
+--
+module Cardano.Wallet.Write.Tx.Balance.CoinSelection
```

Comment:
Alternatively, since the migration-related `TxConstraints` is here too:
```suggestion
module Cardano.Wallet.Write.Tx.CoinSelection
```

Reply:
However, I do kind of like "hiding it" in `CoinSelection` rather than `SizeEstimation`.

Reply:
I would vote for something like `Balance.SizeAndCost` here, since these functions are:
- primarily related to size and cost estimation.
- only incidentally related to coin selection.
- internal to `balanceTx`.

Alternatively, we could have two modules:
- `Balance.Size`
- `Balance.Cost`

Where the `Cost` depends on `Size`.